jsonrpc/server.go

```diff
@@ -316,28 +333,36 @@ func (s *Server) handleRequest(req *request) (*response, error) {
 	return res, nil
 }
 
-func (s *Server) buildArguments(params, handler any, configuredParams []Parameter) ([]reflect.Value, error) {
-	args := make([]reflect.Value, 0, len(configuredParams))
+func (s *Server) buildArguments(ctx context.Context, params any, method Method) ([]reflect.Value, error) {
 	if isNil(params) {
-		if len(configuredParams) > 0 {
+		if len(method.Params) > 0 {
 			return nil, errors.New("missing non-optional param field")
 		}
 
-		return args, nil
+		return make([]reflect.Value, 0), nil
```

Comment:
Hmm, I think we actually had a bug here.

From the spec;

```
params
    A Structured value that holds the parameter values to be used during the invocation of the method. This member MAY be omitted. 
```

This makes the check above a bit too strict. If all of `method.Params` are optional, we should let `params` be nil. But now, we only allow `params` to be nil for methods with no parameters.

`params` is optional, if it is not given we should return the default values for each argument. Not an empty list.

Reply:
Nice catch. Maybe we can resolve this in a separate PR? Seems like a separate issue if I'm understanding correctly.